File: Benchmarks/enterprise/python/flask/testcode/benchmark_test_05905.py
# SPDX-License-Identifier: Apache-2.0
from flask import request, jsonify


def BenchmarkTest05905():
    cookie_value = request.cookies.get('session_token', '')
    data = '%s' % (cookie_value,)
    return jsonify({'status': 'ok'}), 200, {'X-Echo': str(data)}

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest05905.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest05905 {

    private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(BenchmarkTest05905.class);

    @PostMapping(path="/BenchmarkTest05905", consumes="text/plain")
    public void BenchmarkTest05905(@RequestBody String rawBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String rawData = rawBody != null ? rawBody : "";
        String data;
        try { data = String.valueOf(Integer.parseInt(rawData)); }
        catch (NumberFormatException e) { data = rawData; }
        LOG.info("Action: {}", data);
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}
